packages/map-template/src/components/LegendDialog/LegendDialog.scss

```diff
@@ -0,0 +1,76 @@
+.background {
+    z-index: 10000;
+    background-color: var(--color-black-black);
+    opacity: 0.5;
+    position: fixed;
+    width: 100%;
+    height: 100%;
+    top: 0px;
+    left: 0px;
+    overflow: hidden;
+}
+
+.legend {
+    position: absolute;
+    background: white;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    z-index: 100;
+    border-radius: var(--rounding-large);
+    padding: var(--spacing-medium);
+    z-index: 10000;
+    display: grid;
+    width: 500px;
+
+    &__button {
+        color: var(--color-white-white);
+        border-radius: var(--rounding-large);
+        border: none;
+        font-size: var(--font-size-small);
+        width: 100%;
+        height: var(--spacing-xxx-large);
+        position: relative;
+        margin-top: var(--spacing-medium);
+        font-weight: var(--font-weight-x-large);
+        text-transform: uppercase;
+
+        &:hover {
+            filter: brightness(90%);
+        }
+    }
+
+    &__heading {
+        font-size: var(--spacing-x-large);
+        font-weight: var(--font-weight-xx-large);
+        color: var(--color-gray-90);
+        padding-top: var(--spacing-medium);
+    }
+
+    &__section {
+        display: grid;
+        gap: var(--spacing-medium);
+    }
+
+    &__sections {
+        overflow: scroll;
+
+        &--scrollable {
+            &::before {
+                content: '';
+                position: absolute;
+                bottom: calc(64px + var(--spacing-small));
```

Comment:
Should there be a similar gradient for the top?

Reply:
Hmmm, not according to the Figma design 🤔

Reply:
Weird. We have it for long location detail descriptions 🤷

Reply:
I'll check with UX 👍🏻

Reply:
Did you get an answer?

Reply:
Yess, it should also be on top, so I will implement that 😄

Reply:
Added gradient on the top of the legend dialog, when scrollable.